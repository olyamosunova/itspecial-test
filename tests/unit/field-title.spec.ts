import { mount, createLocalVue, shallowMount } from "@vue/test-utils";
import Vue from "vue";
import Vuex from "vuex";
import FieldTitle from "@/components/field-title/Field-title.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

const field = {
    key: 'name',
    label: 'name',
    sortable: false
};

describe('unit tests for FieldTitle component', () => {
    let actions;
    let getters;
    let store: any;

    beforeEach(() => {
        actions = {
            sortingData: jest.fn()
        };

        getters = {
            activeSortBy: () => 'id',
        };

        store = new Vuex.Store({
            actions,
            getters
        });
    });

    it('should match the snapshot', () => {
        const wrapper = mount(FieldTitle, { store, localVue, propsData: { field } });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('change active sort type when user click sort in sortable field', () => {
        field.sortable = true;

        const wrapper = shallowMount(FieldTitle, {
            store,
            localVue,
            propsData: { field }
        });

        expect((wrapper.vm as any).activeSortType).toBe('');
        wrapper.find('th').trigger('click');
        expect((wrapper.vm as any).activeSortType).toBe('inc');
    });

    it('not change active sort type when user click sort in not sortable field', () => {
        field.sortable = false;

        const wrapper = shallowMount(FieldTitle, {
            store,
            localVue,
            propsData: { field }
        });

        expect((wrapper.vm as any).activeSortType).toBe('');
        wrapper.find('th').trigger('click');
        expect((wrapper.vm as any).activeSortType).toBe('');
    });
});
