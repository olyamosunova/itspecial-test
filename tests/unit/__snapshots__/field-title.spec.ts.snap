// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for FieldTitle component should match the snapshot 1`] = `
<th scope="col" class="">
  name
  <!---->
</th>
`;
